hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/snapshot/OMSnapshotCreateRequest.java

```diff
@@ -104,6 +104,8 @@ public OMSnapshotCreateRequest(OMRequest omRequest) {
   @RequireSnapshotFeatureState(true)
   public OMRequest preExecute(OzoneManager ozoneManager) throws IOException {
     final OMRequest omRequest = super.preExecute(ozoneManager);
+    // verify snapshot limit
+    ozoneManager.getOmSnapshotManager().snapshotLimitCheck();
```

Comment:
We should move this after all other validations. Otherwise, inflightCount will never be decreased if snapshot name validation or permission check fails.